File: app/src/main/java/de/tomcory/heimdall/persistence/database/entity/Connection.java
package de.tomcory.heimdall.persistence.database.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Index;

@Entity(primaryKeys = {"appPackage", "hostname"}, indices = {@Index(value = "appPackage"), @Index(value = "hostname")})

public class Connection {

    @NonNull
    public String appPackage;
    @NonNull
    public String hostname;

    public Connection(@NonNull String appPackage, @NonNull String hostname) {
        this.appPackage = appPackage;
        this.hostname = hostname;
    }
}

File: app/src/main/java/de/tomcory/heimdall/persistence/database/entity/Connection.kt
package de.tomcory.heimdall.persistence.database.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Connection(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val protocol: String,
    val initialTimestamp: Long,
    val initiator: String,
    val localPort: Int,
    val remoteHost: String = "",
    val remoteIp: String,
    val remotePort: Int,
    val bytesOut: Long = 0,
    val bytesIn: Long = 0
)
